taichi/program/state_flow_graph.cpp

```diff
@@ -848,7 +969,7 @@ bool StateFlowGraph::optimize_dead_store() {
 
   std::unordered_set<int> to_delete;
   // erase empty blocks
-  for (int i = 1; i < (int)nodes_.size(); i++) {
+  for (int i = first_pending_task_index_; i < (int)nodes_.size(); i++) {
```

Comment:
It's better to make use of `get_pending_tasks()`, but I make the changeset minimal here to avoid conflicts with #1907.

Reply:
sorry, i merged #1907 and it has created mrege conflict now.. But is it possible to use `get_pending_tasks()` now?